infra/gcp/ensure-e2e-projects.sh

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+#
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates & configures projects intended to be used for e2e
+# testing of kubernetes and managed by boskos
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+. "${SCRIPT_DIR}/lib.sh"
+
+function usage() {
+    echo "usage: $0 [repo...]" > /dev/stderr
+    echo "example:" > /dev/stderr
+    echo "  $0 # do all projects" > /dev/stderr
+    echo "  $0 k8s-infra-node-e2e-project # just do one" > /dev/stderr
+    echo > /dev/stderr
+}
+
+## setup service accounts and ips for the prow build cluster
+
+# TODO: replace prow-build-test with actual service account
+PROW_BUILD_SVCACCT=$(svc_acct_email "kubernetes-public" "prow-build-test")
+ensure_service_account \
+  "kubernetes-public" \
+  "prow-build-test" \
+  "used by prowjobs that run in prow-build-test cluster"
+empower_ksa_to_svcacct \
+  "kubernetes-public.svc.id.goog[test-pods/prow-build]" \
+  "kubernetes-public" \
+  "${PROW_BUILD_SVCACCT}"
+
+# manual parts: 
+# - create key, add to prow-build-test as service-account secret
+# - gsutil iam ch serviceAccount:$PROW_BUILD_SVCACCT:objectAdmin gs://bashfire-prow
+# - gsutil iam ch serviceAccount:$PROW_BUILD_SVCACCT:objectCreator gs://bashfire-prow
+
+# TODO: replace boskos-janitor-test with actual service account
+BOSKOS_JANITOR_SVCACCT=$(svc_acct_email "kubernetes-public" "boskos-janitor-test")
+ensure_service_account \
+  "kubernetes-public" \
+  "boskos-janitor-test" \
+  "used by boskos-janitor in prow-build-test cluster"
+empower_ksa_to_svcacct \
+  "kubernetes-public.svc.id.goog[test-pods/boskos-janitor]" \
```

Comment:
why "test-pods" ?

Reply:
comes from a setting in prow: https://github.com/kubernetes/test-infra/blob/master/config/prow/config.yaml#L73 and the fact that boskos is assumed to run in this specific namespace by some jobs and/or the same namespace as the jobs that use it

discussed with the prow team and longer term we'd prefer to refactor prow to support a namespace per build cluster, instead of assuming the same namespace across all build cluster

Reply:
My comment got dis-associated - please comment somewhere nearby so it's less confusing later :)

Reply:
commented before each `empower_ksa_to_svcacct` call